Src/Newtonsoft.Json/JsonConvert.cs

```diff
@@ -969,7 +969,7 @@ public static void PopulateObject(string value, object target, JsonSerializerSet
             }
         }
 
-#if !(NET20 || NET35 || PORTABLE40)
+#if HAVE_TPL
```

Comment:
Can HAVE_TPL be changed to HAVE_ASYNC?

edit: Actually I can see why this is needed. .NET 4.0 supports Task but not async.